platform/platform_linux.cpp

```diff
@@ -268,8 +268,15 @@ void Platform::GetSystemFontNames(FilesList & res) const
 // static
 time_t Platform::GetFileCreationTime(std::string const & path)
 {
+// In older Linux versions there is no reliable way to get file creation time.
+#if __GLIBC__ >= 2 && __GLIBC_MINOR__ >= 28  
   struct statx st;
   if (0 == statx(AT_FDCWD, path.c_str(), 0, STATX_BTIME, &st))
     return st.stx_btime.tv_sec;
+#else
+  struct stat st;
+  if (0 == stat(path.c_str(), &st))
+    return std::min(st.atime, std::min(st.st_mtime, st.st_ctime));
```

Comment:
```suggestion
    return st.st_mtim.tv_sec;
```

`ctime` and `atime` can not be older than `m_time`.

Also I wonder why in `platform_android.cpp` `atime` is used and not `mtime`?

(note that field names are e.g. `st_atim`, not `st_atime`)

Reply:
> 0. What was the compilation error?

```
/home/pastk/dev/organicmaps/platform/platform_linux.cpp: In static member function ‘static time_t Platform::GetFileCreationTime(const string&)’:
/home/pastk/dev/organicmaps/platform/platform_linux.cpp:279:24: error: ‘struct stat’ has no member named ‘atime’
     return std::min(st.atime, std::min(st.st_mtime, st.st_ctime));
                        ^~~~~
```

>     2. Android:
> 
> 
> > The Access time would be the creation date time if the partition is mounted with option noatime, which is the factory default for all 3 of my android phones. That is to say, practically, you can check file/folder creation time.

Yeap you're right `atime` could be older than `mtime` in `noatime` case.
Still, `ctime` can not be older, so no point in checking it.

Reply:
As you've seen, atime may be earlier. 

ctime may also be earlier than mtime and atime (if it's enabled), because it changes only when file's metadata changes, not the content. 

I'm still convinced that my proposed implementation is more robust.

Reply:
`ctime` is updated too when file's contents are changed.

Reply:
Ok, double-checked it. Agree, let's remove ctime check.

Reply:
```suggestion
    return std::min(st.atime, st.st_mtime);
```